### DISCUSSION TITLE
adding url to fields inside split screws up layout

### DISCUSSION THREAD

**Discussion by jdion84** [filamentphp/filament#9125] [Help]
when using `->url` on a `TextColumn`, if that column is inside a split, the layout gets screwed up because the actual `a` tag has a tailwind class of `w-full`

is there a workaround for this, or would a PR need submitting to remove that class?

**Comment by leandrocfe** [comment]
Could you share the code please?

**Comment by jdion84** [comment]
sure!

screwed up layout:

```php
$table->columns([
    Split::make([
        TextColumn::make('name'),

        Stack::make([
            TextColumn::make('email')
                ->url(fn ($state) => "mailto:$state"),

            TextColumn::make('phone')
                ->url(fn ($state) => "tel:$state"),
        ]),
    ]),
])
```

image: https://i.imgur.com/W65Shfl.png

perfect layout:

```php
$table->columns([
    Split::make([
        TextColumn::make('name'),

        Stack::make([
            TextColumn::make('email'),

            TextColumn::make('phone'),
        ]),
    ]),
])
```

image: https://i.imgur.com/0Fnee0O.png

as you can see, its pushing things over due to what i'm assuming is the `w-full` class of the link

even if i add `grow(false)`, it doesn't make a difference, because this class is hard-coded in the source code

i have also noticed this screws up when using `Grid` as well, especially on mobile

**Comment by jdion84** [comment]
@leandrocfe tagging so you see my reply

**Reply by leandrocfe** [reply]
Ok let me try...

**Reply by leandrocfe** [reply]
@jdion84 try it please
```php
TextColumn::make('email')
    ->formatStateUsing(fn ($state): string => "<a href=\"mailto:{$state}\">{$state}</a>")
    ->html()
```

**Reply by jdion84** [reply]
that is broken as well, it aligns it to the center and wraps every word

**Reply by leandrocfe** [reply]
Hum, I think you could create a [custom column](https://filamentphp.com/docs/3.x/tables/columns/custom#custom-classes) and use divs + tailwind to achieve this result.
